File: backend/app/services/timetable/period_schedule.py
"""Fixed school schedule constants.

The school day is no longer configurable:
  - Start time: 07:10
  - End time:   12:35
  - Period duration: 40 minutes 37.5 seconds (40.625 min)
  - Lunch period: always the 4th period

325 minutes / 40.625 min = 8 teaching periods per day exactly.
"""

from typing import Dict, List

PERIODS_PER_DAY: int = 8
LUNCH_PERIOD: int = 4
START_TIME: str = "07:10"
END_TIME: str = "12:35"
PERIOD_DURATION_MINUTES: float = 40.625

# Each period's start/end derived from START_TIME + n * 40.625 minutes.
# Displayed as HH:MM (seconds/seconds-fraction truncated) per the API contract.
PERIOD_SCHEDULE: List[Dict[str, object]] = [
    {"period": 1, "start_time": "07:10", "end_time": "07:50"},
    {"period": 2, "start_time": "07:50", "end_time": "08:31"},
    {"period": 3, "start_time": "08:31", "end_time": "09:11"},
    {"period": 4, "start_time": "09:11", "end_time": "09:52"},
    {"period": 5, "start_time": "09:52", "end_time": "10:33"},
    {"period": 6, "start_time": "10:33", "end_time": "11:13"},
    {"period": 7, "start_time": "11:13", "end_time": "11:54"},
    {"period": 8, "start_time": "11:54", "end_time": "12:35"},
]

File: frontend/src/features/timetable/periodSchedule.ts
// Fixed school schedule — mirrors the backend constant in
// backend/app/services/timetable/period_schedule.py. The school day is no
// longer configurable: start 07:10, 8 periods/day, lunch in period 4.
// Each period is 40.625 min; times are shown as HH:MM.
export const PERIODS_PER_DAY = 8;
export const LUNCH_PERIOD = 4;

export interface PeriodTime {
  period: number;
  start: string;
  end: string;
}

export const PERIOD_TIMES: PeriodTime[] = [
  { period: 1, start: '07:10', end: '07:50' },
  { period: 2, start: '07:50', end: '08:31' },
  { period: 3, start: '08:31', end: '09:11' },
  { period: 4, start: '09:11', end: '09:52' },
  { period: 5, start: '09:52', end: '10:33' },
  { period: 6, start: '10:33', end: '11:13' },
  { period: 7, start: '11:13', end: '11:54' },
  { period: 8, start: '11:54', end: '12:35' },
];

export const getPeriodTimeStr = (periodNum: number): string => {
  const p = PERIOD_TIMES.find(t => t.period === periodNum);
  if (!p) return '';
  return `${p.start} – ${p.end}`;
};

export const getPeriodStartTime = (periodNum: number): string | null =>
  PERIOD_TIMES.find(t => t.period === periodNum)?.start ?? null;
